packages/x-charts/src/BarChart/BarElement.tsx

```diff
@@ -145,17 +145,16 @@ function BarElement(props: BarElementProps) {
     elementType: Bar,
     externalSlotProps: slotProps?.bar,
```

Comment:
This should work and allow you to not spread `other` below:

```suggestion
    externalSlotProps: slotProps?.bar,
    externalForwardedProps: other,
```

It should also automatically merge the event handlers and stuff like `ref` and `style`

Reply:
This is a nice suggestion! 👍 
Other than that, the changes look great! 💯

Reply:
I've not done the other components, because the TS result seems buggy. even if `other` type contains `d: string` the resulting props do not contain it

Reply:
The typing of `useSlotProps` is ... complex :laughing: